### ISSUE TITLE
Is there a way to skip Napoleon extension for some part of the source code?

### ISSUE THREAD

**Issue by mshonichev** [sphinx-doc/sphinx#11808]
### Describe the bug

I use Napoleon extension to generate documentation about python sources.

However, I got vendored portion of code within a source base which I can not modify. And this code does not follow Google docstring style guide, which causes errors during documentation generation.

Is there any simple possibility to skip `Napoleon` for some packages/folders and fallback to regular autodoc or skip autodoc for that code altogether?

### How to Reproduce

`my_code.py`
```
class Foo:
    def bar(a: str) -> None:
    """
    Creates a bar.

    Args:
            a (str): a param
    Raises:
            Exception: whenever needed
    """
```

`vendor_code.py`
```
from pydantic import BaseModel, Field
class Vendor(BaseModel):
    attr: str | None = Field(
        default=None,
        example="some vendor example",
    )
    """
    [Some nasty] string with : and all other stuff`  with e.g. broken links https:// or altogether Returns: breaking docs
    """
```

### Environment Information

```text
Platform:              darwin; (macOS-11.4-x86_64-i386-64bit)
Python version:        3.11.6 (main, Oct 11 2023, 11:34:27) [Clang 12.0.5 (clang-1205.0.22.9)])
Python implementation: CPython
Sphinx version:        7.2.6
Docutils version:      0.20.1
Jinja2 version:        3.1.2
Pygments version:      2.17.2
```

### Sphinx extensions

```python
extensions = [
    "sphinx.ext.autodoc",
    "sphinx.ext.doctest",
    "sphinx.ext.napoleon",
]

source_suffix = {
    ".rst": "restructuredtext",
}
```

**Comment by electric-coder** [comment]
@mshonichev write a custom extension in your `conf.py` using [`autodoc-skip-member`](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#event-autodoc-skip-member). There are a couple of good examples on SO, see [A 5599712](https://stackoverflow.com/a/5599712) and [A 3757500](https://stackoverflow.com/q/3757500).

Also please notice that Sphinx support question [should be asked on Stack Overflow](https://www.sphinx-doc.org/en/master/internals/contributing.html#get-help) with GitHub being preferred for issue and feature requests.

**Comment by picnixz** [comment]
TL;DR: not natively.

Alternatively, I'd suggest you link directly to the vendored code instead of generating documentation for that part. Like, "see [vendored] part". Then, you can exclude the whole vendor code from ever being parsed.

And yes, questions should be asked on SO instead of here. We are however planning to open the "Discussions" tab for that at some point (I cannot guarantee anything on when).
